### ISSUE TITLE
Exception recursion

### ISSUE THREAD

**Issue by ParanoikCZE** [Magenic/MAQS#69]
We experience very long logs in case of exceptions. These usually contain like 1000+ repeats of the same exception details, e.g. 
```
no such element: Unable to locate element: {"method":"css selector","selector":"*selector*"}
  (Session info: chrome=75.0.3770.90)
   at OpenQA.Selenium.Remote.RemoteWebDriver.UnpackAndThrowOnError(Response errorResponse)
   at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Remote.RemoteWebDriver.FindElement(String mechanism, String value)
   at OpenQA.Selenium.Support.Events.EventFiringWebDriver.FindElement(By by)
   at Magenic.Maqs.BaseSeleniumTest.Extensions.Wait.<>c__DisplayClass38_0.<ElementIsVisible>b__0(IWebDriver driver)
   at OpenQA.Selenium.Support.UI.DefaultWait`1.Until[TResult](Func`2 condition)
```
Based on my analysis, exception is catched here https://github.com/SeleniumHQ/selenium/blob/master/dotnet/src/support/Events/EventFiringWebDriver.cs#L1528 then throws and fires OnExceptionEvent, on which MAQS registers loging https://github.com/Magenic/MAQS/blob/master/Framework/BaseSeleniumTest/SeleniumDriverManager.cs#L281 here https://github.com/Magenic/MAQS/blob/master/Framework/BaseSeleniumTest/SeleniumDriverManager.cs#L182

I'm totally lost where this recursion comes from.

**Comment by TroyWalshProf** [comment]
Couple of questions to make sure we are answering the real issue

1. Which version of MAQS are you on?
2. What is your logging level: https://magenic.github.io/MAQS/#/MAQS_5/EnterpriseConfiguration?id=logging-levels
3. Is part of the first line missing, I would expect to see something like this: `VERBOSE:	Exception thrown: OpenQA.Selenium.NoSuchElementException: no such element: Unable to locate element: {"method":"css selector","selector":"#jdjd"}`

**Comment by TroyWalshProf** [comment]
My guess is that you are using verbose logging, which will print all the retries in a fluent wait.

**Comment by TroyWalshProf** [comment]
Sorry, did not mean to close

**Comment by ParanoikCZE** [comment]
Hi, thanks for your comments.
1) 5.5
2) Information (we used default setup from wiki)
3) there is no first part, not sure where is that lost, but maybe Nunit throw it away during log parsing
4) Verbose (hope it's close to information) logging makes a perfect sense after you described that. I'm going to lower it to warning.
Will ping you back after check.

**Comment by TroyWalshProf** [comment]
Would it be possible to get a sanitized copy of one of the files and the app.config?
It does appear we are doing double logging for NUnit test, but I want to make sure that issue you are seeing.

Side note* 
We are looking at revamping the logging so txt logs are a bit cleaner and html logs are prettier (more userful). More coming on that hopefully by the end of the year.

**Comment by ParanoikCZE** [comment]
I'm not sure, how Nunit catches these logs, I know it works if we output to console, but I didn't find any for MAQS/Selenium. Can we set a private channel for sending these?

**Comment by TroyWalshProf** [comment]
No need, I found the issue:
https://github.com/Magenic/MAQS/blob/457fafae09c46ee04594991c7c46d860610bcbab/Framework/BaseTest/BaseTest.cs#L488

Will have it fixed in the next release, which should be by the end of month

**Comment by TroyWalshProf** [comment]
@ParanoikCZE One quick note*
If you have not done so already you want to switch over to [Magenic.Maqs](https://www.nuget.org/packages/Magenic.Maqs/) if you are using Magenic.Open.Maqs.
We recently open sourced our entire testing framework (Magenic.Maqs) so Magenic.Open.Maqs is being depricated.  - AKA Open MAQS will no longer be getting updates

**Comment by ParanoikCZE** [comment]
Hi, thanks for info, seems like there are still few weird issues (e.g. failed KillDriver method call reports few dozens of times), but it's much better now, so let's wait for release and we'll see.
And yep, we already switched to MM some time ago.

**Comment by FermJacob** [comment]
@ParanoikCZE This is fixed in MAQS 5.7 and has been released to NuGet.  Can you re-test the issue and see if the logging issues are fixed?

**Comment by FermJacob** [comment]
@ParanoikCZE Any update if this is resolved?

**Comment by ParanoikCZE** [comment]
> @ParanoikCZE Any update if this is resolved?

Buddy, it's holidays, slow down :) 
(working on it, will update results)

**Comment by ParanoikCZE** [comment]
Unfortunately, I can't see any improvement ;(
Logs folder now contains usually like 3 files per test in a format like
Testname - 2020-01-03-04-33-42-7391.txt
Testname - 2020-01-03-04-34-29-6345.txt
Testname - 2020-01-03-04-34-29-6502.txt
While the first one contains 3 exceptions, second one 2 and last one only single one but with "ERROR: Test failed".
Interesting thing is that NUnit, while capturing console output, logged this exception 13 times.

```ERROR: 
no such element: Unable to locate element: {"method":"css selector","selector":"***consored***"}
(Session info: chrome=75.0.3770.90)
at OpenQA.Selenium.Remote.RemoteWebDriver.UnpackAndThrowOnError(Response errorResponse)
at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
at OpenQA.Selenium.Remote.RemoteWebDriver.FindElement(String mechanism, String value)
at OpenQA.Selenium.Support.Events.EventFiringWebDriver.FindElement(By by)
at Magenic.Maqs.BaseSeleniumTest.Extensions.Wait.<>c__DisplayClass38_0.<ElementIsVisible>b__0(IWebDriver driver)
at OpenQA.Selenium.Support.UI.DefaultWait`1.Until[TResult](Func`2 condition)

```

**Comment by TroyWalshProf** [comment]
I am not able to repo what you are seeing on my end.
Would it be possible to get some simplified sample code?
*If it helps you can have it point to http://magenicautomation.azurewebsites.net/ so no private business info gets included.

**Comment by TroyWalshProf** [comment]
Oh, you can also suppress first chance exceptions:
https://magenic.github.io/MAQS/#/MAQS_5/EnterpriseConfiguration?id=first-chance-exception-handler-optional

I personally feel like it cuts too much but not very one wants the level of detail I want in my tests.

**Comment by ParanoikCZE** [comment]
At first, sorry for delay, we've got some fire to solve. 
Secondly .. I've realized that we rerun most of testcases more times with different variables. So I tried to reproduce it this way and got it. So, if you execute test method which is in format 
Test([Values(1, 2)] int run)
which fails on some MAQS/Selenium exception, exception is being aggregated and reported every time for every previous execution. So in second run, I get two same errors and so on.
Similar way, we execute the whole classes with different users via TestFixture attribute, so the problem rises with every other run.
However, this is only observable in console output when executed via NUnit console.
If still interested, I can prepare a simple solution to present this. Just ping me prefered way to receive this.
Btw we don't use UseFirstChanceHandler property in our config, so should be in default.
